src/sdl/main.cpp

```diff
@@ -1,18 +1,24 @@
-#include <SDL2/SDL.h>
-#include <stdio.h>
+//
+// NOTE: form include-what-you-use reported diagnostics:
+//
+#include <SDL.h>          // for SDL_Quit
+#include <SDL_audio.h>    // for SDL_GetAudioDriver, SDL_GetNumAudioDrivers
+#include <SDL_error.h>    // for SDL_GetError
+#include <SDL_pixels.h>   // for SDL_MapRGB
+#include <SDL_surface.h>  // for SDL_FillRect, SDL_Surface
+#include <SDL_timer.h>    // for SDL_Delay
+#include <SDL_version.h>  // for SDL_version, SDL_GetVersion
+#include <SDL_video.h>    // for SDL_CreateWindow, SDL_DestroyWindow, SDL_Ge...
+
+#include <stdio.h>        // for fprintf, stderr
+#include <string.h>       // for strcmp
+#include <cstdlib>        // for NULL
```

Comment:
IIRC, this file was copied directly from an SDL example; that's why it used the C version of `stdio.h`. 

If we are modernizing for C++, I think we really need to use the C++ versions of these headers: `<cstdio>` and `<cstring>`.

Also, if `NULL` is the only thing we need from `<cstdlib>`, we could remove that header and replace all the `NULL`s with `nullptr`. `nullptr` has been a part of the core language since C++11, and doesn't need to be defined by a macro in a library.

Reply:
I noticed that `include-what-you-use` recommends often the `c` header instead the `c++` header variant.

I think, I will split this MR into different parts next week. 

I am waiting for the cmakelib patch first.